lib/playlistGenerator/Playlist.js

```diff
@@ -437,19 +437,18 @@ Playlist.prototype.recalculateOffsetsAndDuration = function recalculateOffsetsAn
 
     var retVal = new TimeRange();
 
-    var minMaxInfos = _.map(that.inner.durations, function (d, index){
+    var minMaxInfos = _.map(that.inner.durations, (d, index) => {
         this.inner.durations[index].value = 0;
         return calcClipStartTimeAndDuration.call(this,index);
-    },that).filter(function(minMax){
-        return minMax.valid;
     });
 
-    if(minMaxInfos.length) {
+    if(minMaxInfos.length && _.every(minMaxInfos,r=>r.valid)) {
         retVal.min = minMaxInfos.first.min;
         retVal.max = minMaxInfos.last.max;
-        that.updateGap();
     }
 
+    that.updateGap();
```

Comment:
check  that taking the call to updateGap from the condition cannot cause any new bug.